spec/requests/auth/sessions_spec.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Auth Sessions' do
+  describe 'POST /auth/sign_in' do
+    # The rack-attack check has issues with the non-nested invalid param used here
+    before { Rack::Attack.enabled = false }
+    after { Rack::Attack.enabled = true }
```

Comment:
What kind of issues?

Reply:
When this line is hit - https://github.com/mastodon/mastodon/blob/main/config/initializers/rack_attack.rb#L141 - the `dig` there fails because (in this spec, with the purposefully invalid param) the first one is a string not a hash. So, for purposes of this spec, seemed easier to disable. Alternately, could update the RA config to handle different inbound data as well?

Reply:
Sounds sensible, thanks for the explanation. Erroring on invalid parameters is not too much of an issue, I don't think it warrants making the rack-attack config more complex.